src/persist/src/indexed/mod.rs

```diff
@@ -454,6 +547,14 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
             }
         }
 
+        for (id, new_future_ts_lower) in future_ts_lower_updates {
```

Comment:
maybe add a comment here indicating "now that all of the trace writes / metadata writes are committed, its safe to truncate future, and it also doesn't matter if truncating future fails that can be best effort (it doesnt prompt a rollback)"